Add theme support for tinted-nvim

JamyGolden opened the pull request:
This PR adds support for [tinted-nvim](https://github.com/tinted-theming/tinted-nvim) using the `base16` theme value.

shadmansaleh commented on lua/lualine/themes/base16.lua:
```diff
@@ -90,6 +90,34 @@ local function setup_base16_nvim()
+  local loaded, tinted = pcall(require, 'tinted-colorscheme')
+
+  if not loaded then
+    add_notice(
```
This is problematic as this will show this notice even when base16_nvim is present

shadmansaleh commented:
Also, you can put the theme under `/lua/lualine/themes/` in the colorschemes repository too. Even if you have the same filename, base16.lua` lualine guarantees it'll load the theme from your repo instead of from itself. That way, when users have your colorscheme installed, your theme gets loaded in lualine. And you have full control over the colors without having to wait for merging updates here.

JamyGolden commented:
Yeah tinted-vim is the continuation of base16-vim, where as tinted-nvim (new) is a fork of base16-nvim which has been extended. I contacted the base16-nvim maintainer about combining our efforts into 1 repo but they said they didn't have capacity so ended up being a fork.

A lot of nvim users like using a plugin which is built with lua and they tend to pick something other than tinted-vim since it's built with vimscript. We have a lot of people using tinted-vim who don't use nvim, so we ended up having both under the org with different people maintaining the different repos.

JamyGolden commented:
I've added a `.../themes/tinted.lua` file and added support for tinted-nvim and tinted-vim plugin and added info to the README. I've removed all changes I made to the `.../themes/base16.lua` file.

JamyGolden commented:
Can you link me to an example of a theme doing this? I'm not quite grasping the setup if it's in the tinted-nvim repo. How is the theme imported and used in lualine if it's external?

shadmansaleh commented:
> Can you link me to an example of a theme doing this?

Most of the colorschemes actually do that already. For example [onedark](https://github.com/navarasu/onedark.nvim/blob/master/lua/lualine/themes/onedark.lua) [tokyonight](https://github.com/folke/tokyonight.nvim/blob/main/lua/lualine/themes/tokyonight.lua)...

> I'm not quite grasping the setup if it's in the tinted-nvim repo. How is the theme imported and used in lualine if it's external?

You don't need to do anything special. You can just place the same file in same location in your repo. The [loading](https://github.com/nvim-lualine/lualine.nvim/blob/15884cee63a8c205334ab13ab1c891cd4d27101a/lua/lualine/utils/loader.lua#L208-L254) is handled by us. You don't need to worry about that. You basically just need to export the theme in the lualines format from file in lua/lualine/themes/theme-name.lua . You are already doing that in the file here. No need to do anything different.

JamyGolden commented:
Ah ok interesting! How often does lualine pull those lua theme files from the different repos? I've merged a PR on tinted-nvim: https://github.com/tinted-theming/tinted-nvim/pull/15 

I've updated this PR to just include README info for the `tinted` theme option. (I guess it should be merged after lualine pulls the theme file from tinted-nvim repo)

shadmansaleh commented:
> How often does lualine pull those lua theme files from the different repos?

We don't : ] 

You're still misunderstanding. Basically, we will load them from your repo at runtime. Lets say someone has both tinted-nvim and lualine installed.  And are using tinted-nvim as their colorscheme. Lualine will load that theme file from tinted-nvims installation. It doesn't need to belong in this repo. Also as long as you name the theme file same as your colorschemes name user doesn't even need to explicitly configure. It get's auto infered when default "auto" theme is used. Basically theme file exisiting in lualines repo and in any other repo is the same to lualine. It actually prioritizes external ones. Say you keep a base64.lua theme in your repo lualine will load that one instead of the one in this repo. We kind of don't even merge new themes nowadays. The ones that are currently there have historically been there.

JamyGolden commented:
Haha oh ok I see now, all dynamically handled, very nice! Thanks for all the info!
